x/mongo/driver/topology/pool.go

```diff
@@ -825,12 +861,58 @@ func (p *pool) checkInNoEvent(conn *connection) error {
 	return nil
 }
 
+// clearAll does same as the "clear" method and interrupts all in-use connections as well.
+func (p *pool) clearAll(err error, serviceID *primitive.ObjectID) {
+	p.clearImpl(err, serviceID, true)
+}
+
+func (p *pool) interruptInUseConnections() {
+	for _, conn := range p.conns {
+		if conn.inUse && p.stale(conn) {
+			_ = conn.closeWithErr(poolClearedError{
```

Comment:
Optional: Consider inverting this logic and de-indenting the code below.

E.g.
```go
if !conn.inUse || !p.stale(conn) {
    continue
}

_ = conn.closeWithErr(...)
...
```
```